app/models/mixins/authentication_mixin.rb

```diff
@@ -266,9 +266,13 @@ def authentication_check(*args)
 
   private
 
+  def build_verify_args(type, options)
+    kind_of?(Host) ? [type, options] : type
+  end
+
   def authentication_check_no_validation(type, options)
     header  = "type: [#{type.inspect}] for [#{id}] [#{name}]"
-    verify_args = self.kind_of?(Host) ? [type, options] : type
+    verify_args = build_verify_args(type, options)
```

Comment:
I like the removal of this coupling to `Host`.

Actually I think we can always pass the options to `verify_credentials`.
The only test that is failing is `spec/models/mixins/authentication_mixin_spec.rb:338` which can be easily fixed